### PR TITLE
rpc: add `Client::wait_until_healthy` method

### PR DESCRIPTION
**Pull request by tony-iqlusion**
Adds a method which repeatedly polls the `/health` endpoint until it returns a successful response or a specified timeout elapses.

This is useful when writing tests that need to wait for a node to boot up (e.g. inside of a docker container), e.g.:

https://github.com/cosmos/cosmos-rust/pull/71/files#diff-0e31f23fc223d1d543169a682e0f1093c1cc53c6a312febce7d5d1854919fafaR84-R96

### CODE REVIEW

File: rpc/src/client.rs [hunk 1]
```diff
@@ -221,6 +224,31 @@ pub trait Client {
...
+    /// the given `timeout` has elapsed.
+    async fn wait_until_healthy<T>(&self, timeout: T) -> Result<()>
+    where
+        T: Into<Duration> + Send,
+    {
+        let timeout = timeout.into();
+        let poll_interval = Duration::from_millis(200);
+        let mut attempts_remaining = timeout.as_millis() / poll_interval.as_millis();
+
+        while let Err(_) = self.health().await {
```
Comment (thanethomson) [inline rpc/src/client.rs:237]:
```suggestion
        while self.health().await.is_err() {
```

Quick fix for the clippy suggestion.

**Review by thanethomson — COMMENTED** [review]
This is a great idea! Have you tested it locally on your side?

**Comment by tony-iqlusion** [comment]
I updated the PR linked in the description to use code that is now nearly identical to this and it's working as expected locally:

https://github.com/cosmos/cosmos-rust/pull/71/files#diff-0e31f23fc223d1d543169a682e0f1093c1cc53c6a312febce7d5d1854919fafaR97-R117

**Review by thanethomson — COMMENTED** [review]
Awesome. I'd like to include this in the v0.19.0 release. Could you please add this to the `CHANGELOG` under the `IMPROVEMENTS` section?

**Comment by thanethomson** [comment]
Actually don't worry about it - I'll add something tomorrow when I finalize the release. It'll be easier that way so we don't have merge conflicts in the changelog.

**Comment by tony-iqlusion** [comment]
I noticed one small problem with this trying to use it.

After updating to `tendermint-rpc` v0.19, I was still using `tcp://` URLs, and even though those were failing to parse, this code waited the full timeout and just said it timed out, making it a bit hard to debug.

Perhaps it should look for certain kinds of errors, i.e. ones relating to the remote node being down, and bubble all others up.

Otherwise I switched my code to use it and it's working great!

**Comment by thanethomson** [comment]
> After updating to tendermint-rpc v0.19, I was still using tcp:// URLs, and even though those were failing to parse, this code waited the full timeout and just said it timed out, making it a bit hard to debug.

Yes, I think I've just found a small bug in the `rpc::Url` -> `uri::Uri` conversion that would cause this to happen when specifying a `tcp://` address. I'll open an issue :+1:

**Pull request merged**